components/aws/sagemaker/batch_transform/component.yaml

```diff
@@ -4,78 +4,102 @@ description: |
     default: '{}'
+    type: JSON
   - name: input_location
     description: 'The S3 location of the data source that is associated with a channel.'
+    type: String
   - name: data_type
     description: 'Data type of the input. Can be ManifestFile, S3Prefix, or AugmentedManifestFile.'
     default: 'S3Prefix'
+    type: String
   - name: content_type
     description: 'The multipurpose internet mail extension (MIME) type of the data.'
     default: ''
+    type: String
   - name: split_type
     description: 'The method to use to split the transform job data files into smaller batches.'
     default: 'None'
+    type: String
   - name: compression_type
     description: 'If the transform data is compressed, the specification of the compression type.'
     default: 'None'
+    type: String
   - name: output_location
     description: 'The Amazon S3 path where you want Amazon SageMaker to store the results of the transform job.'
+    type: String
   - name: accept
     description: 'The MIME type used to specify the output data.'
     default: ''
+    type: String
   - name: assemble_with
     description: 'Defines how to assemble the results of the transform job as a single S3 object. Either None or Line.'
     default: ''
+    type: String
   - name: output_encryption_key
     description: 'The AWS Key Management Service ID of the key used to encrypt the output data.'
     default: ''
+    type: String
   - name: input_filter
     description: 'A JSONPath expression used to select a portion of the input data to pass to the algorithm.'
     default: ''
+    type: String
   - name: output_filter
     description: 'A JSONPath expression used to select a portion of the joined dataset to save in the output file for a batch transform job.'
     default: ''
+    type: String
   - name: join_source
     description: 'Specifies the source of the data to join with the transformed data.'
     default: 'None'
+    type: String
   - name: instance_type
     description: 'The ML compute instance type.'
     default: 'ml.m4.xlarge'
+    type: String
   - name: instance_count
     description: 'The number of ML compute instances to use in each training job.'
     default: '1'
+    type: Integer
   - name: resource_encryption_key
     description: 'The AWS KMS key that Amazon SageMaker uses to encrypt data on the storage volume attached to the ML compute instance(s).'
     default: ''
+    type: String
   - name: endpoint_url
     description: 'The endpoint URL for the private link VPC endpoint.'
     default: ''
+    type: String
   - name: tags
     description: 'Key-value pairs to categorize AWS resources.'
     default: '{}'
+    type: JSON
 outputs:
   - {name: output_location,    description: 'S3 URI of the transform job results.'}
 implementation:
   container:
-    image: amazon/aws-sagemaker-kfp-components:0.3.0
-    command: ['python']
+    image: redbackthomson/aws-kubeflow-sagemaker:20200501-1
```

Comment:
Was it intentional to use a non-Amazon repo?

Reply:
This is not the final resting spot for the code - it will go into the amazon Dockerhub repository before this code is merged. As part of our development cycle we need to push it to Dockerhub to run the unit and integration tests, and from there we will deploy it and update this PR.

Reply:
Can we optimize the flow? This will be unnecessary change. If you finish testing and code change, you can fetch master with this PR change to build the image. 

@RedbackThomson

Reply:
Right. That's basically what I ended up doing. Are you saying I should not post the intermediate container to the PR? I should just update the image to the final version?

Reply:
I think we should not post intermediate container just update here the final version of image.